configure: Allow overriding uname results

In a cross compile setting it makes no sense to rely on the "uname" values
reported by the build host system. This patch allows overriding the
"uname -r", "uname -s" and "uname -m" results with the "UNAME_R", "UNAME_S"
and "UNAME_M" environment variables.

Signed-off-by: Jo-Philipp Wich <jo@mein.io>

--- a/configure.ac
+++ b/configure.ac
@@ -354,7 +354,7 @@ Setting up SunOS kernel module(s)"
         fi
     }
 
-    release=`uname -r`
+    release=${UNAME_R:-`uname -r`}
     karch=`/usr/bin/isainfo -k`
     makext="sol2"
     archvariant=
